### ISSUE TITLE
no-script-url: add option to allow `javascript:void(0)`

### ISSUE THREAD

**Issue by unframework** [eslint/eslint#10744]
**What rule do you want to change?**

`no-script-url`

**Does this change cause the rule to produce more or fewer warnings?**

Fewer, if option is enabled

**How will the change be implemented? (New option, new default behavior, etc.)?**

New option: `allowVoidZero`

**Please provide some example code that this change will affect:**

```jsx
// valid with the option enabled
<form action="javascript:void(0)" onSubmit={() => doSomething()}>
</form>

// still invalid with the option enabled
<a href="javascript:alert('hello')">eval link</a>
```

**What does the rule currently do for this code?**

Currently, `javascript:void(0)` strings are flagged as invalid code because it is indeed technically a form of `eval`.

**What will the rule do after it's changed?**

With the option enabled, the `javascript:void(0)` string (but no other `javascript:`-prefixed string) would be treated as an idiomatic "no-op" that it is, commonly used for things like forms and links.

Of note: for links people often use `#`, but it still requires to do `event.preventDefault()` to avoid address bar change. For forms, `#` definitely requires `event.preventDefault()` because otherwise form contents are appended directly to the URL. No-op URLs help skip that boilerplate a bit and are fairly resilient.

Happy to work on a PR if this makes sense to the maintainers! Thanks, by the way, for your hard work on the project.

**Comment by platinumazure** [comment]
I'm in favor, that's a very common pattern. Thanks for the issue!

**Comment by bendtherules** [comment]
Would like to have this too. Have always disabled this cases manually till now.

**Comment by platinumazure** [comment]
I'll champion this.

@eslint/eslint-team Thoughts on this proposal?

**Comment by s4san** [comment]
`javascript:;` is also a common pattern. I think that should be white listed too.

**Comment by unframework** [comment]
@platinumazure - I understand that this is still going through the team buy-in process and not set in stone, but I can probably "optimistically" get started on a PR?

**Comment by platinumazure** [comment]
@unframework Absolutely- there is just the risk of you wasting your time if we end up not accepting the proposal. But you are more than welcome to start a PR. Thanks!

**Comment by kaicataldo** [comment]
I see a thumbs down without a comment from @michaelficarra. Mind explaining why you’re against this, for the sake of discussion?

**Comment by platinumazure** [comment]
@michaelficarra @mysticatea Why are you opposed to this proposal?

**Comment by platinumazure** [comment]
It looks like we weren't able to obtain consensus around this issue from the ESLint team. I'm going to close this issue. This doesn't mean the issue wasn't a good idea: it's just something the team cannot commit to implementing at this time. Thanks for contributing regardless, we appreciate it!

**Comment by xiaotiandada** [comment]
orz
